lib/ansible/modules/cloud/amazon/aws_neptune_subnet_group.py

```diff
@@ -0,0 +1,187 @@
+        )
+    except client.exceptions.from_code('DBSubnetGroupNotFound'):
+        return {'exists': False}
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't verify existence of database subnet group")
+
+    return {'current_config': response['DBSubnetGroups'][0], 'exists': True}
+
+
+def create_group(client, module, params):
+    try:
+        response = client.create_db_subnet_group(**params)
+        get_waiter(
+            client, 'subnet_group_available'
+        ).wait(
+            DBSubnetGroupName=params['DBSubnetGroupName']
+        )
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't create database subnet group")
+
+    return {'db_subnet_group_arn': response['DBSubnetGroup']['DBSubnetGroupArn'], 'changed': True}
+
+
+def update_group(client, module, params, group_status):
+    param_changed = []
+    param_keys = list(params.keys())
+    current_keys = list(group_status['current_config'].keys())
+    common_keys = set(param_keys) - (set(param_keys) - set(current_keys))
+    for key in common_keys:
+        if (params[key] != group_status['current_config'][key]):
+            param_changed.append(True)
+        else:
+            param_changed.append(False)
+
+    if any(param_changed):
+        try:
+            response = client.modify_db_subnet_group(**params)
+            get_waiter(
+                client, 'subnet_group_available'
+            ).wait(
+                DBSubnetGroupName=params['DBSubnetGroupName']
+            )
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, msg="Couldn't update database subnet group")
+        return {'db_subnet_group_arn': response['DBSubnetGroup']['DBSubnetGroupArn'], 'changed': True}
+    else:
+        return {'db_subnet_group_arn': group_status['current_config']['DBSubnetGroupArn'], 'changed': False}
+
+
+def delete_group(client, module, params):
+    try:
+        response = client.delete_db_subnet_group(
+            DBSubnetGroupName=params['DBSubnetGroupName']
+        )
+        get_waiter(
+            client, 'subnet_group_deleted'
+        ).wait(
+            DBSubnetGroupName=params['DBSubnetGroupName']
+        )
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't delete database subnet group")
+
+    return {'db_subnet_group_arn': '', 'changed': True}
+
+
+def main():
+    module = AnsibleAWSModule(
+        argument_spec={
+            'name': dict(type='str', required=True),
+            'state': dict(type='str', choices=['present', 'absent'], default='present'),
+            'description': dict(type='str', required=True),
+            'subnet_ids': dict(type='list', required=True),
+        },
+        supports_check_mode=True,
+    )
+
```

Comment:
Right now earlier botocore versions traceback. Since this is such a new version of botocore there should be a check here so the user gets a helpful message.
```
if not module.botocore_at_least('1.10.30'):
    module.fail_json(msg="This module requires botocore >= 1.10.30")
```